webdriver-spec.html

```diff
@@ -5340,26 +5402,146 @@ <h3 id="character-types">Character types</h3>
+  <td>
+   <a>Accept Alert</a><br />
+   <a>Dismiss Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
 
-<section>
-<h3>window.alert, window.prompt and window.confirm</h3>
+ <tr>
+  <td><dfn>Confirm</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-confirm><code>confirm</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
+
+ <tr>
+  <td><dfn>Prompt</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-prompt><code>prompt</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a><br />
+   <a>Send Alert Text</a>
+ </tr>
+</table>
+
+<p class=issue>It’s not clear what interoperability looks like
+ for the ability to switch top-level browsing contexts
+ whilst a user prompt is present.
+
+<p>The <dfn>current user prompt</dfn> is said to be the active
+ <a href=https://html.spec.whatwg.org/#user-prompts>user prompt</a>,
+ which can be one of the entries on the <a>table of simple dialogues</a>,
+ the window.<a href=https://html.spec.whatwg.org/#dom-print><code>print</code></a> dialogue,
+ or a dialogue spawned by <a href=https://html.spec.whatwg.org/#dom-showmodaldialog><code>showModalDialog()</code></a>
+ should this be supported by the user agent.
+
+<p>To <dfn data-lt="dismiss|dismissed|dismisses">dismiss</dfn>
+ the <a>current user prompt</a>,
+ do so as if the user would click the <b>Cancel</b> or <b>OK</b> button,
+ whichever is present, in that order.
+
+<p>To <dfn data-lt="accept|accepted|accepts">accept</dfn> the <a>current user prompt</a>,
+ do so as if the user would click the <b>OK</b> button.
+
+<p>A <dfn>user prompt handler</dfn> is
+ an <a href=https://html.spec.whatwg.org/#enumerated-attribute>enumerated attribute</a>
+ defining what action the <a>remote end</a> must take
+ when a <a href=https://html.spec.whatwg.org/#user-prompts>user prompt</a> is encountered.
+ The following table lists the keywords and states for the attribute:
+
+<table class=simple>
+ <tr>
+  <th>Keyword
+  <th>State
+  <th>Description
+ </tr>
 
-<p>When the remote end has a modal dialog
- such as those produced from <code>window.alert</code>,
- <code>window.prompt</code> and <code>window.confirm</code>
- it SHOULD allow the interrogation of the dialogues using the following interface.
+ <tr>
+  <td>"<code>dismiss</code>"
+  <td><dfn>Dismiss state</dfn>
+  <td>All <a href=https://html.spec.whatwg.org/#simple-dialogs>simple dialogues</a>
+   encountered should be <a>dismissed</a>.
+ </tr>
+
+ <tr>
+  <td>"<code>accept</code>"
+  <td><dfn>Accept state</dfn>
+  <td>All <a href=https://html.spec.whatwg.org/#simple-dialogs>simple dialogues</a>
+   encountered should be <a>accepted</a>.
+ </tr>
+</table>
+
+<p>In order to <dfn>handle any user prompts</dfn>
+ a <a>remote end</a> must take the following steps:
+
+<ol>
+ <li><p>If there is no <a>current user prompt</a>,
```

Comment:
This whole definition is meant for auto-handling prompts, yes.  Basically the match below look for a _user prompt handler_ and if there isn’t one (_missing value default state_), it will return an _unexpected alert open_ error.